rct229/rulesets/ashrae9012019/section10/section10rule14.py

```diff
@@ -0,0 +1,510 @@
+                        eff
+                        for eff, metric in modeled_effs_b
+                        if metric == expected_low_temp_eff_metric_b
+                    ),
+                    None,
+                )
+
+            return {
+                "hvac_system_type_b": hvac_system_type_b,
+                "total_capacity_b": total_capacity_b,
+                "is_zone_agg_factor_undefined_and_needed": is_zone_agg_factor_undefined_and_needed,
+                "expected_effs_b": expected_effs_b,
+                "modeled_effs_b": modeled_effs_b,
+                "expected_high_temp_eff_b": expected_high_temp_eff_b,
+                "modeled_high_temp_eff_b": modeled_high_temp_eff_b,
+                "expected_low_temp_eff_b": expected_low_temp_eff_b,
+                "modeled_low_temp_eff_b": modeled_low_temp_eff_b,
+            }
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
+            hvac_system_type_b = calc_vals["hvac_system_type_b"]
+            expected_effs_b = calc_vals["expected_effs_b"]
+            expected_high_temp_eff_b = calc_vals["expected_high_temp_eff_b"]
+            expected_low_temp_eff_b = calc_vals["expected_low_temp_eff_b"]
+            modeled_effs_b = calc_vals["modeled_effs_b"]
+            modeled_high_temp_eff_b = calc_vals["modeled_high_temp_eff_b"]
+            modeled_low_temp_eff_b = calc_vals["modeled_low_temp_eff_b"]
+            total_capacity_b = calc_vals["total_capacity_b"]
+            is_zone_agg_factor_undefined_and_needed = calc_vals[
+                "is_zone_agg_factor_undefined_and_needed"
+            ]
+
+            # Case 3 and 7 both satisfied here
+            if (
+                len(modeled_effs_b) > 0
+                and all(
+                    modeled_eff_b[0] == expected_eff_b[0]
+                    for modeled_eff_b, expected_eff_b in zip(
+                        modeled_effs_b, expected_effs_b
+                    )
+                )
+                and total_capacity_b is None
+            ):
+                return True
+
+            # Case 8
+            if (
+                hvac_system_type_b == HVAC_SYS.SYS_4
+                and modeled_high_temp_eff_b is not None
+                and modeled_high_temp_eff_b == expected_high_temp_eff_b
+                and modeled_low_temp_eff_b is not None
+                and modeled_low_temp_eff_b == expected_low_temp_eff_b
+                and total_capacity_b is None
+            ):
+                return True
+
+            # Case 4 and 10 both satisfied here
+            if (
+                len(modeled_effs_b) > 0
+                and all(
+                    modeled_eff_b[0] == expected_eff_b[0]
+                    for modeled_eff_b, expected_eff_b in zip(
+                        modeled_effs_b, expected_effs_b
+                    )
+                )
+                and is_zone_agg_factor_undefined_and_needed
+            ):
+                return True
```

Comment:
I think we can condense the logic more but I do appreciate the clarity in the code so it is OK to keep as it is.
If you think this can be improved a little more without losing the clarify, I suggest considering:
```python
if (
                len(modeled_effs_b) > 0
                and all(
                    modeled_eff_b[0] == expected_eff_b[0]
                    for modeled_eff_b, expected_eff_b in zip(
                        modeled_effs_b, expected_effs_b
                    )
                )
                and (
                        #Case 3 and 7 both satisfied here
                        total_capacity_b is None 
                        or
                        # Case 4 and 10 both satisfied here
                        is_zone_agg_factor_undefined_and_needed
                )
```